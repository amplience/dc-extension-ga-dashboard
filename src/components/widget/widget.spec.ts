import { render } from '@testing-library/svelte';
import Widget from './widget.svelte';

describe('Widget', () => {
  it('should render the Widget component', () => {
    const { container } = render(Widget, { loading: false });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the Widget component with loading spinner', () => {
    const { container } = render(Widget, { loading: true });

    expect(container.firstChild).toMatchSnapshot();
  });
});
