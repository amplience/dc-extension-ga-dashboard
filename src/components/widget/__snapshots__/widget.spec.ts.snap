// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget should render the Widget component 1`] = `
<div>
  <article
    class="widget svelte-pntwh0"
  >
     
  </article>
</div>
`;

exports[`Widget should render the Widget component with loading spinner 1`] = `
<div>
  <article
    class="widget svelte-pntwh0"
  >
    <div
      class="spinner-container svelte-12lddhy"
      style="--z-index: 10;"
    >
      <div
        class="spinner dot-0 svelte-12lddhy"
      />
      <div
        class="spinner dot-1 svelte-12lddhy"
      />
      <div
        class="spinner dot-2 svelte-12lddhy"
      />
    </div>
     
  </article>
</div>
`;
